cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -47,8 +47,9 @@ def vm_name():
 def instance_args(request, provider, provisioning, vm_name):
     """ Fixture to prepare instance parameters for provisioning
     """
-    inst_args = dict(template_name=provisioning.get('image', {}).get('image') or provisioning.get(
+    inst_args = dict(template_name=provisioning.get('image', {}).get('name') or provisioning.get(
         'template'))
+    assert inst_args.get('template_name')
```

Comment:
@jaryn I think if `None` will come it will assert error. As per me its setup where we are taking `template` name from `provisioning` l.e. provider data.
I think better to `skip` if what you think?

Reply:
I disagree, and that's not the pattern we've applied to the test modules. When a necessary configuration value is not available, the test is skipped. The user can't know ahead of time that they explicitly don't want to run this test, how would they know this and trigger a skip?

Ideally the tests run with no errors - just failures and skips. I especially don't want to see errors in fixtures (even AssertionError) when we can easily see and error that could happen.

`provisioning` fixture is just giving us the AttrDict from cfme_data for the provider - and it itself will `pytest.skip` if that section is missing.

If we're then trying to lookup more detailed keys in provisioning, then we should skip if they're missing. We just don't have the requisite data for the provider that was parametrized for the test.

The fact that these keys are required for these tests - the provider marker should be using a more detailed ProviderFilter with `required_fields`. This will exclude providers from collection before we even get to this point.

Reply:
Ok. As you wish.